python/sidecar/http.py

```diff
@@ -28,62 +29,192 @@
+    response = chat.ask_question(req)
     return response
 
 
 @sidecar_api.post("/search")
-async def search(req: SearchRequest) -> SearchResponse:
-    response = search_module.search_s2(req)
-    return response
\ No newline at end of file
+async def http_search(req: SearchRequest) -> SearchResponse:
+    response = search.search_s2(req)
+    return response
+
+
+# Project API
+# --------------
+@project_api.get("/")
+async def list_projects():
+    """
+    Returns a list of projects for the current user
+    """
+    user_id = "user1"
+    projects_dict = projects.read_project_path_storage(user_id)
+    return projects_dict 
+
+
+@project_api.post("/")
+async def create_project(project_path: str = None):
+    """
+    Creates a project directory in the filesystem
+
+    Parameters
+    ----------
+    project_path : str
+        The path to the project directory. Only necessary for Desktop.
+        For web, the project is stored in a private directory on the server.
+    """
+    user_id = "user1"
+    project_id = str(uuid4())
+    project_path = projects.create_project(user_id, project_id, project_path)
+    return {
+        project_id: project_path,
+    }
+
+
+@project_api.get("/{project_id}")
+async def get_project(project_id: str):
+    """
+    Returns the project path and a list of files in the project
+    """
+    user_id = "user1"
+    project_path = projects.get_project_path(user_id, project_id)
+    filepaths = projects.get_project_files(user_id, project_id)
+    return {
+        "project_id": project_id,
+        "project_path": project_path,
+        "filepaths": filepaths,
+    }
+
+
+@project_api.delete("/{project_id}")
+async def delete_project(project_id: str):
+    """
+    Deletes a project directory and all files in it
+    """
+    user_id = "user1"
+    projects.delete_project(user_id, project_id)
+    return {
+        "status": "success",
+        "message": "Project deleted",
+        "project_id": project_id,
+    }
+
+
+@project_api.get("/{project_id}/files")
+async def get_project_files(project_id: str):
+    user_id = "user1"
+    filepaths = projects.get_project_files(user_id, project_id)
+    return filepaths
+
+
+
+# Filesystem API
+# --------------
+@filesystem_api.put("/{project_id}/{filepath:path}")
+async def create_file(project_id: str, filepath: Path, file: UploadFile = File(...)):
+    user_id = "user1"
+    project_path = projects.get_project_path(user_id, project_id)
+    filepath = project_path / filepath
+
+    if not filepath.exists():
+        filepath.parent.mkdir(parents=True, exist_ok=True)
+    
+    try:
+        with open(filepath, "wb") as f:
+            shutil.copyfileobj(file.file, f)
+    except Exception as e:
+        print(e)
+    finally:
+        file.file.close()
+    return {
+        "status": "success",
+        "message": "File uploaded",
+        "filepath": filepath,
+    }
+
+
+@filesystem_api.get("/{project_id}/{filepath:path}")
```

Comment:
Is there anything that prevents reading from outside the project path here? Could `filepath` be something like `../../../some-other-folder` or `/etc/passwd`? Just wondering about security here.